app/models/flowchart.rb

```diff
@@ -1,4 +1,43 @@
 # frozen_string_literal: true
 
 class Flowchart < ApplicationRecord
+  validates :title, format: { with: /[a-zA-Z ]/ }, null: false
+  validates :description, format: { with: /[a-zA-Z ]/ }, null: false
+  validates :height, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, null: false
+  validates :deleted, inclusion: { in: [true, false] }, null: false
+
+  def calculate_and_set_max_height
```

Comment:
Is this method intended to be called whenever we add a FlowchartNode to the Flowchart? Would it make sense to call this whenever a request is made to get a Flowchart (i.e. in `all_flowcharts` and `serialized_flowchart_by_id)? 

The latter would result in less calls to `calculate_and_set_max_height`, but it would also slow down the requests for retrieving flowchart information.

Reply:
I like the latter.